test/test_exceptions.py

```diff
@@ -45,3 +46,16 @@ def test_header_parsing_errors(self):
 
         assert "defects" in str(hpe)
         assert "unparsed_data" in str(hpe)
+
+
+class TestNewConnectionError:
+    def test_pool_property_deprication_warning(self):
+        err = NewConnectionError(HTTPConnectionPool("localhost"), "test")
+        with pytest.warns(DeprecationWarning) as record:
+            err.pool
```

Comment:
So we're doing something with this property can we `assert err.pool is err.conn`?